index.js

```diff
@@ -28,14 +29,14 @@ const fsReadFileAsync = promisify(fs.readFile);
 
 class HtmlWebpackPlugin {
   /**
-   * @param {Partial<HtmlWebpackPluginOptions>} [options]
+   * @param {HtmlWebpackOptions} [options]
```

Comment:
Could we find a better name here?  
Maybe `HtmlWebpackPluginOptionsWithDefaults` or something similar?

Reply:
That's not what's getting passed in here though. This is the constructor argument which is the `Options` from the type definition file. Lines 60/61 is where the "options with defaults" part is setup which (which I named `HtmlWebpackInternalOptions`)

Reply:
You are right - I meant `HtmlWebpackInternalOptions`

Reply:
`HtmlWebpackPluginConfiguration`?

I'm not a fan of `HtmlWebpackOptionsWithDefaults` as the interface doesn't actually have defaults.

Reply:
But actually it holds the `UserOptions` + `defaults` or is that wrong?

Reply:
It's just an interface that specifies that all fields are required + a change in how `minify` should look. The interface is actually used to describe both the `defaults` and the `user options` + `defaults` given both use cases need to have all fields as required.

Reply:
What about `ProcessedHtmlWebpackOptions` ?

Reply:
sure. I just made the change